lightning/src/util/events.rs

```diff
@@ -567,6 +592,37 @@ pub enum Event {
 		/// now + 5*time_forwardable).
 		time_forwardable: Duration,
 	},
+	/// Used to indicate that we've intercepted an HTLC forward. This event will only be generated if
+	/// you've encoded an intercept scid in the receiver's invoice route hints using
+	/// [`ChannelManager::get_intercept_scid`] and have set [`UserConfig::accept_intercept_htlcs`].
+	///
+	/// [`ChannelManager::forward_intercepted_htlc`] or
+	/// [`ChannelManager::fail_intercepted_htlc`] MUST be called in response to this event. See
+	/// their docs for more information.
+	///
+	/// [`ChannelManager::get_intercept_scid`]: crate::ln::channelmanager::ChannelManager::get_intercept_scid
+	/// [`UserConfig::accept_intercept_htlcs`]: crate::util::config::UserConfig::accept_intercept_htlcs
+	/// [`ChannelManager::forward_intercepted_htlc`]: crate::ln::channelmanager::ChannelManager::forward_intercepted_htlc
+	/// [`ChannelManager::fail_intercepted_htlc`]: crate::ln::channelmanager::ChannelManager::fail_intercepted_htlc
+	HTLCIntercepted {
+		/// The fake scid that was programmed as the next hop's scid, generated using
+		/// [`ChannelManager::get_intercept_scid`].
+		///
+		/// [`ChannelManager::get_intercept_scid`]: crate::ln::channelmanager::ChannelManager::get_intercept_scid
+		requested_next_hop_scid: u64,
+		/// The payment hash used for this HTLC.
+		payment_hash: PaymentHash,
+		/// How many msats were received on the inbound edge of this HTLC.
+		inbound_amount_msat: u64,
+		/// How many msats the payer intended to route to the next node. Depending on the reason you are
```

Comment:
Hmm, my text editor is telling me it already is

Reply:
Hmmm ok I think it's because my editor treats tabs as 4 chars. I'm at column 105 here 😅. Just out of curiosity for future PRs on my side: Do we treat tabs as one character only?

Reply:
I don't think we have a rule for that, just roughly 100 lol. I'm holding out for rustfmt here

Reply:
new rust style team pls save us